### PR TITLE
Add an ubsan-enabled job

### PR DESCRIPTION
**Pull request by jvoisin**

### CODE REVIEW

**Comment by struct** [comment]
This is failing because its overriding CFLAGS. I think if we add something like this specifically for CLI flags it should work fine:

```
diff --git a/Makefile b/Makefile
index 9b46578..c275325 100644
--- a/Makefile
+++ b/Makefile
@@ -225,7 +225,7 @@ CFLAGS += $(COMMON_CFLAGS) $(SECURITY_FLAGS) $(BUILD_ERROR_FLAGS) $(HOOKS) $(HEA
        -std=c11 $(SANITIZER_SUPPORT) $(ALLOC_SANITY) $(MEMCPY_SANITY) $(UNINIT_READ_SANITY) $(CPU_PIN) $(SCHED_GETCPU) \
        $(EXPERIMENTAL) $(UAF_PTR_PAGE) $(VERIFY_BIT_SLOT_CACHE) $(NAMED_MAPPINGS) $(ABORT_ON_NULL) $(NO_ZERO_ALLOCATIONS) \
        $(ABORT_NO_ENTROPY) $(ISO_DTOR_CLEANUP) $(SHUFFLE_BIT_SLOT_CACHE) $(USE_SPINLOCK) $(HUGE_PAGES) $(USE_MLOCK) \
-       $(MEMORY_TAGGING) $(STRONG_SIZE_ISOLATION) $(MEMSET_SANITY)
+       $(MEMORY_TAGGING) $(STRONG_SIZE_ISOLATION) $(MEMSET_SANITY) $(CLI_FLAGS)
 CXXFLAGS = $(COMMON_CFLAGS) -DCPP_SUPPORT=1 -std=c++17 $(SANITIZER_SUPPORT) $(HOOKS)
 EXE_CFLAGS = -fPIE
 GDB_FLAGS = -g -ggdb3 -fno-omit-frame-pointer
```

File: .github/workflows/testsuite.yml [hunk 1]
```diff
@@ -70,6 +70,12 @@ jobs:
       - run: apk update && apk add build-base bash musl-dev
       - run: make tests CC=gcc CXX=g++
       - run: make cpp_tests CC=gcc CXX=g++
+  testsuite-gcc-ubsan:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: CC=gcc CXX=g++ CFLAGS=-fsanitize=undefined make tests
```
Comment (struct) [inline .github/workflows/testsuite.yml:77]: These vars need to go after the `make` command and should include `$CLI_CFLAGS`

Comment (jvoisin) [reply .github/workflows/testsuite.yml:77]: Nope, because they're environment variables here :) (I think.)

Comment (struct) [reply .github/workflows/testsuite.yml:77]: They have to be passed after the make command as they are variables to make:

```
root@6c85958e84c0:/code/isoalloc# CC=gcc CXX=g++ CFLAGS=-fsanitize=undefined make tests  
rm -rf build/* tests_perf_analysis.txt big_tests_perf_analysis.txt gmon.out test_output.txt *.dSYM core* iso_alloc_profiler.data
rm -rf android/libs android/obj
mkdir -p build/
make library_debug_unit_tests
**clang** -fsanitize=undefined -Wall -Iinclude/ -DTHREAD_SUPPORT=1 -pthread -DPRE_POPULATE_PAGES=1 -DSMALL_MEM_STARTUP=0 -DSANITIZE_CHUNKS=0 -DFUZZ_MODE=0 -DPERM_FREE_REALLOC=0 -DDISABLE_CANARY=0 -DNEVER_REUSE_ZONES=0 -Wno-pointer-arith -Wno-gnu-zero-variadic-macro-arguments -Wno-format-pedantic -Werror -pedantic -DMALLOC_HOOK=1  -fvisibility=hidden -std=c11     -DALLOC_SANITY=0 -DMEMCPY_SANITY=0 -DUNINIT_READ_SANITY=0 -DCPU_PIN=0 -DSCHED_GETCPU -DEXPERIMENTAL=0 -DUAF_PTR_PAGE=0 -DVERIFY_BIT_SLOT_CACHE=0 -DNAMED_MAPPINGS=0 -DABORT_ON_NULL=0 -DNO_ZERO_ALLOCATIONS=1 -DABORT_NO_ENTROPY=1 -DISO_DTOR_CLEANUP=0 -DSHUFFLE_BIT_SLOT_CACHE=0 -DUSE_SPINLOCK=0 -DHUGE_PAGES=1 -DUSE_MLOCK=1 -DMEMORY_TAGGING=0 -DSTRONG_SIZE_ISOLATION=0 -DMEMSET_SANITY=0 -fPIC -shared  -DUNIT_TESTING=1 -DDEBUG=1 -DLEAK_DETECTOR=1 -DMEM_USAGE=1 -g -ggdb3 -fno-omit-frame-pointer src/*.c -o build/libisoalloc.so
^Cmake: *** [Makefile:299: tests] Interrupt

root@6c85958e84c0:/code/isoalloc# make tests CC=gcc CXX=g++ CFLAGS=-fsanitize=undefined
rm -rf build/* tests_perf_analysis.txt big_tests_perf_analysis.txt gmon.out test_output.txt *.dSYM core* iso_alloc_profiler.data
rm -rf android/libs android/obj
mkdir -p build/
make library_debug_unit_tests
**gcc -fsanitize=undefined** -fPIC -shared  -DUNIT_TESTING=1 -DDEBUG=1 -DLEAK_DETECTOR=1 -DMEM_USAGE=1 -g -ggdb3 -fno-omit-frame-pointer src/*.c -o build/libisoalloc.so

```

**Pull request closed without merging**